src/INode.java

```diff
@@ -388,13 +407,17 @@ private ASTValue evalQueueSay(WordsEnvironment environment) {
 		WordsObject object;
 		if (referenceObject.type.equals(ValueType.OBJ)){
 			WordsProperty property = referenceObject.objValue.getProperty(identifier.stringValue);
-			// TODO: Check that property is in fact an object, or throw an exception
+			if (property.type != WordsProperty.PropertyType.OBJECT) {
+				throw new InvalidTypeException(this.lineNo, ValueType.OBJ.toString(), property.type.toString());
+			}
 			object = property.objProperty;
 		} else {
 			object = environment.getObject(identifier.stringValue);
 		}
 		
-		//TODO: If message is not a string, throw an exception
+		if (message.type != ValueType.STRING) {
```

Comment:
I realized my original comment was slightly wrong.  The reference manual allows either a string or a number (which is then coerced to a string).  So this should be changed to coerce a number to a string, or just open an issue about that point.